src/azure-cli-core/azure/cli/core/commands/client_factory.py

```diff
@@ -102,6 +102,34 @@ def configure_common_settings(cli_ctx, client):
     client.config.generate_client_request_id = 'x-ms-client-request-id' not in cli_ctx.data['headers']
 
 
+def configure_common_settings_track2(cli_ctx):
+    client_kwargs = {}
+
+    client_kwargs.update(_debug.change_ssl_cert_verification_track2())
+
+    client_kwargs['logging_enable'] = True
+    client_kwargs['user_agent'] = get_az_user_agent()
+
+    try:
+        command_ext_name = cli_ctx.data['command_extension_name']
+        if command_ext_name:
+            client_kwargs['user_agent'] += "CliExtension/{}".format(command_ext_name)
+    except KeyError:
+        pass
+
+    headers = dict(cli_ctx.data['headers'])
+    command_name_suffix = ';completer-request' if cli_ctx.data['completer_active'] else ''
+    headers['CommandName'] = "{}{}".format(cli_ctx.data['command'], command_name_suffix)
+    if cli_ctx.data.get('safe_params'):
+        headers['ParameterSetName'] = ' '.join(cli_ctx.data['safe_params'])
+    client_kwargs['headers'] = headers
+
+    if 'x-ms-client-request-id' in cli_ctx.data['headers']:
+        client_kwargs['request_id'] = cli_ctx.data['headers']['x-ms-client-request-id']
```

Comment:
Some questions here:
1. `request_id` is required for client?
2. Should we use `x-ms-request-id`?

Reply:
Actually  `request_id` will be set to  "x-ms-client-request-id" in request header, you can check the code in azure core below:

def on_request(self, request):
        # type: (PipelineRequest) -> None
        """Updates with the given request id before sending the request to the next policy.

        :param request: The PipelineRequest object
        :type request: ~azure.core.pipeline.PipelineRequest
        """
        request_id = unset = object()
        if 'request_id' in request.context.options:
            request_id = request.context.options.pop('request_id')
            if request_id is None:
                return
        elif self._request_id is None:
            return
        elif self._request_id is not _Unset:
            request_id = self._request_id   # type: ignore
        elif self._auto_request_id:
            request_id = str(uuid.uuid1())  # type: ignore
        if request_id is not unset:
            header = {"x-ms-client-request-id": request_id}

---
In reply to: [407376852](https://github.com/Azure/azure-cli/pull/12952#discussion_r407376852) [](ancestors = 407376852)